library/Requests.php

```diff
@@ -205,20 +204,47 @@ protected static function get_transport($capabilities = array()) {
 				continue;
 			}
 
-			$result = call_user_func(array($class, 'test'), $capabilities);
+			$result = $class::test($capabilities);
 			if ($result) {
 				self::$transport[$cap_string] = $class;
-				break;
+				return $class;
 			}
 		}
-		if (self::$transport[$cap_string] === null) {
+
+		return '';
+	}
+
+	/**
+	 * Get a working transport
+	 *
+	 * @throws Requests_Exception If no valid transport is found (`notransport`)
+	 * @return Requests_Transport
+	 */
+	protected static function get_transport($capabilities = array()) {
+		$class = self::get_transport_class($capabilities);
+
+		if ($class === '') {
```

Comment:
Is the check against an empty string sufficient here ? Or should this check that the `$class` is not listed in the array of valid transports ? (code hardening; check for what you want, not for what you don't want)

Reply:
We don't have a list of valid transports that includes possible extensions.

The empty string is the "error case" that is being returned by `get_transport_class()`, I used an empty string so that I can keep a strong type of `string`.